[WK2] Apply method signature validation to all types of IPC message handling

zdobersek opened the pull request:
#### c16b84bd344f2188fa548498b29c2b42a8448345
<pre>
[WK2] Apply method signature validation to all types of IPC message handling
<a href="https://bugs.webkit.org/show_bug.cgi?id=248975">https://bugs.webkit.org/show_bug.cgi?id=248975</a>

Reviewed by Kimmo Kinnunen.

Validation of CompletionHandler argument types in IPC&apos;s message-handling
code is reworked and also applied to the messages&apos; argument types.

The MethodSignatureValidation template takes a message-handling method&apos;s
function type and works through the argument types, deducing the
argument types specified by the method as well as any CompletionHandler
object that&apos;s used in messages that require a reply. When done, the
template instantiation will provide a std::tuple&lt;&gt; alias of those
argument types as well as a std::tuple&lt;&gt; alias of argument types of any
CompletionHandler object.

The different IPC::handleMessage() variants now use those instantiations
to perform compile-time asserts, ensuring the desired message-handling
method works with the same types as those specified by the IPC message.
In case of messages with a reply, the CompletionHandler&apos;s argument types
are similarly compared to the IPC message&apos;s reply argument types.

This enables some simplifications in the messages.py generator. Some IPC
messages and message-handling methods also need to be adjusted, either
adopting argument types that perfectly match the current code, or in
some cases using a sized integer type.

WebPage::loadRequestWaitingForProcessLaunch() was so far using the
NO_RETURN macro to apply the noreturn attribute through the C-style
__attribute__ specifier. This caused some trouble with certain compilers
since that attribute is then ingrained into the function type, which
was then causing trouble in the MethodSignatureValidation template
specializations. To avoid that but also apply the necessary no-return
semantics to the method, the C++11-style [[noreturn]] attribute is used.

* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
(WebKit::RemoteDisplayListRecorder::drawSystemImage):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.h:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::mergeStatisticForTesting):
(WebKit::NetworkProcess::insertExpiredStatisticForTesting):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/Platform/IPC/HandleMessage.h:
(IPC::MethodSignatureValidation&lt;R):
(IPC::handleMessage):
(IPC::handleMessageWantsConnection):
(IPC::handleMessageSynchronous):
(IPC::handleMessageSynchronousWantsConnection):
(IPC::handleMessageAsync):
(IPC::handleMessageAsyncWantsConnection):
(IPC::CompletionHandlerValidation&lt;CompletionHandler&lt;void): Deleted.
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
(IPC::jsValueForDecodedArguments):
* Source/WebKit/Scripts/webkit/messages.py:
(function_parameter_type):
(message_to_struct_declaration):
(arguments_type): Deleted.
(reply_arguments_type): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments const):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessages.h:
(Messages::TestWithIfMessage::LoadURL::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::SendImageData::arguments const):
(Messages::TestWithImageData::ReceiveImageData::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::LoadURL::arguments const):
(Messages::TestWithLegacyReceiver::LoadSomething::arguments const):
(Messages::TestWithLegacyReceiver::TouchEvent::arguments const):
(Messages::TestWithLegacyReceiver::AddEvent::arguments const):
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments const):
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments const):
(Messages::TestWithLegacyReceiver::Close::arguments const):
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments const):
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments const):
(Messages::TestWithLegacyReceiver::SendInts::arguments const):
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments const):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments const):
(Messages::TestWithLegacyReceiver::GetPlugins::arguments const):
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments const):
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments const):
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments const):
(Messages::TestWithLegacyReceiver::TemplateTest::arguments const):
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments const):
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments const):
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments const):
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments const):
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::SendSemaphore::arguments const):
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessages.h:
(Messages::TestWithStreamBatched::SendString::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessages.h:
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendString::arguments const):
(Messages::TestWithStream::SendStringSync::arguments const):
(Messages::TestWithStream::SendMachSendRight::arguments const):
(Messages::TestWithStream::ReceiveMachSendRight::arguments const):
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::LoadURL::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessage::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments const):
(Messages::TestWithSuperclass::TestSyncMessage::arguments const):
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::LoadURL::arguments const):
(Messages::TestWithoutAttributes::LoadSomething::arguments const):
(Messages::TestWithoutAttributes::TouchEvent::arguments const):
(Messages::TestWithoutAttributes::AddEvent::arguments const):
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments const):
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments const):
(Messages::TestWithoutAttributes::Close::arguments const):
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments const):
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments const):
(Messages::TestWithoutAttributes::SendInts::arguments const):
(Messages::TestWithoutAttributes::CreatePlugin::arguments const):
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments const):
(Messages::TestWithoutAttributes::GetPlugins::arguments const):
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments const):
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments const):
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments const):
(Messages::TestWithoutAttributes::TemplateTest::arguments const):
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments const):
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments const):
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments const):
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments const):
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments const):
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::stopMessageTesting):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPasteboardProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::engineFailedToLoad):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioHardwareListener.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteImageDecoderAVFManager.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadRequestWaitingForProcessLaunch):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/257836@main">https://commits.webkit.org/257836@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c9be270cc27d41a190bbfe4608a02e791898fb34

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100090 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9256 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33257 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109431 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10140 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86758 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92528 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107321 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105858 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7663 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/91024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34372 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/103604 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89569 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3036 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/23934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3001 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Big-Sur-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9133 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/43422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5381 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4847 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
```diff
@@ -317,7 +317,7 @@ void RemoteDisplayListRecorder::drawNativeImageWithQualifiedIdentifier(Qualified
 }
 
-void RemoteDisplayListRecorder::drawSystemImage(SystemImage& systemImage, const FloatRect& destinationRect)
+void RemoteDisplayListRecorder::drawSystemImage(const Ref<SystemImage>& systemImage, const FloatRect& destinationRect)
```
Needing to have `const Ref<T>` in the signature looks like a regression.
This could perhaps be worked around by doing an unrelated improvement described below..

zdobersek replied:
Ref<>&& is more appropriate in general, forgot it's possible.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
```diff
@@ -53,13 +53,13 @@ class SendImageData {
 
 private:
-    Arguments m_arguments;
+    std::tuple<const RefPtr<WebCore::ImageData>&> m_arguments;
```
- `T` becomes `T&`
- `Ref<T>` becomes `const Ref<T>&`

If we instead made Ref, RefPtr, RetainPtr to be pass-by-value into the messages, they would be more natural types and there would not be a double indirection when accessing them.

In fact, the list could contain SharedMemory::Handle, UnixFileDescriptor, etc. too, which would then solve the move implementation issue..

zdobersek replied:
This PR affects decoding, so essentially how the arguments from the message are passed on to the handling method. With this added validation, in its current simplest form the method's parameter types have to mach the types of arguments, but the method still has flexibility of handling each argument through either type of references or just by value. It would be possible to special-case things like `Ref<>`, but I would prefer stricter matching.

For passing objects into the message during encoding, I would avoid special-casing in the generator and give more control for that purpose through how the message  is constructed. So essentially what #6485 tries to do, but a bit more polished, less intrusive, and incrementally adoptable.

kkinnunen-apple reviewed (commented):
the unsigned -> uint64_t seems to indicate this would be useful.

kkinnunen-apple commented on Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
```diff
@@ -318,11 +318,11 @@ void RemoteDisplayListRecorder::drawNativeImageWithQualifiedIdentifier(Qualified
 }
 
-void RemoteDisplayListRecorder::drawSystemImage(SystemImage& systemImage, const FloatRect& destinationRect)
+void RemoteDisplayListRecorder::drawSystemImage(Ref<SystemImage>&& systemImage, const FloatRect& destinationRect)
```
I don't know if it makes a difference, but:

I think for Ref/RefPtr and similar, it's more performant to pass in by value.
Contrast:
The caller does not know whether the callee moves the object off. So caller must provide a ptr pointing to their stack. After the call, caller must destruct the object.

VS.

Te caller knows that they don't need the object. They can construct the object directly into the conceptual callee argument stack.

To visualize it the another way:

consider
```
void f() {
     auto si = getSystemImage();
     drawSystemImage(WTFMove(si);
}
```

The call is in tail-call position. If the drawSystemImage takes the value, the compiler could omit the return to f() and instead return to the caller of f().

If the drawSystemImage takes rvalue ref, the return must happen to f() in order to do the maintainance for possibly non-moved Ref<SystemImage> si.

zdobersek replied:
I see, adopted.

zdobersek committed:
[WK2] Apply method signature validation to all types of IPC message handling
https://bugs.webkit.org/show_bug.cgi?id=248975

Reviewed by Kimmo Kinnunen.

Validation of CompletionHandler argument types in IPC's message-handling
code is reworked and also applied to the messages' argument types.

The MethodSignatureValidation template takes a message-handling method's
function type and works through the argument types, deducing the
argument types specified by the method as well as any CompletionHandler
object that's used in messages that require a reply. When done, the
template instantiation will provide a std::tuple<> alias of those
argument types as well as a std::tuple<> alias of argument types of any
CompletionHandler object.

The different IPC::handleMessage() variants now use those instantiations
to perform compile-time asserts, ensuring the desired message-handling
method works with the same types as those specified by the IPC message.
In case of messages with a reply, the CompletionHandler's argument types
are similarly compared to the IPC message's reply argument types.

This enables some simplifications in the messages.py generator. Some IPC
messages and message-handling methods also need to be adjusted, either
adopting argument types that perfectly match the current code, or in
some cases using a sized integer type.

WebPage::loadRequestWaitingForProcessLaunch() was so far using the
NO_RETURN macro to apply the noreturn attribute through the C-style
__attribute__ specifier. This caused some trouble with certain compilers
since that attribute is then ingrained into the function type, which
was then causing trouble in the MethodSignatureValidation template
specializations. To avoid that but also apply the necessary no-return
semantics to the method, the C++11-style [[noreturn]] attribute is used.

* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp:
(WebKit::RemoteDisplayListRecorder::drawSystemImage):
* Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.h:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::mergeStatisticForTesting):
(WebKit::NetworkProcess::insertExpiredStatisticForTesting):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/Platform/IPC/HandleMessage.h:
(IPC::MethodSignatureValidation<R):
(IPC::handleMessage):
(IPC::handleMessageWantsConnection):
(IPC::handleMessageSynchronous):
(IPC::handleMessageSynchronousWantsConnection):
(IPC::handleMessageAsync):
(IPC::handleMessageAsyncWantsConnection):
(IPC::CompletionHandlerValidation<CompletionHandler<void): Deleted.
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
(IPC::jsValueForDecodedArguments):
* Source/WebKit/Scripts/webkit/messages.py:
(function_parameter_type):
(message_to_struct_declaration):
(arguments_type): Deleted.
(reply_arguments_type): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments const):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessages.h:
(Messages::TestWithIfMessage::LoadURL::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::SendImageData::arguments const):
(Messages::TestWithImageData::ReceiveImageData::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::LoadURL::arguments const):
(Messages::TestWithLegacyReceiver::LoadSomething::arguments const):
(Messages::TestWithLegacyReceiver::TouchEvent::arguments const):
(Messages::TestWithLegacyReceiver::AddEvent::arguments const):
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments const):
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments const):
(Messages::TestWithLegacyReceiver::Close::arguments const):
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments const):
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments const):
(Messages::TestWithLegacyReceiver::SendInts::arguments const):
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments const):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments const):
(Messages::TestWithLegacyReceiver::GetPlugins::arguments const):
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments const):
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments const):
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments const):
(Messages::TestWithLegacyReceiver::TemplateTest::arguments const):
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments const):
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments const):
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments const):
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments const):
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::SendSemaphore::arguments const):
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessages.h:
(Messages::TestWithStreamBatched::SendString::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessages.h:
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendString::arguments const):
(Messages::TestWithStream::SendStringSync::arguments const):
(Messages::TestWithStream::SendMachSendRight::arguments const):
(Messages::TestWithStream::ReceiveMachSendRight::arguments const):
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::LoadURL::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessage::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments const):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments const):
(Messages::TestWithSuperclass::TestSyncMessage::arguments const):
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments const):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::LoadURL::arguments const):
(Messages::TestWithoutAttributes::LoadSomething::arguments const):
(Messages::TestWithoutAttributes::TouchEvent::arguments const):
(Messages::TestWithoutAttributes::AddEvent::arguments const):
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments const):
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments const):
(Messages::TestWithoutAttributes::Close::arguments const):
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments const):
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments const):
(Messages::TestWithoutAttributes::SendInts::arguments const):
(Messages::TestWithoutAttributes::CreatePlugin::arguments const):
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments const):
(Messages::TestWithoutAttributes::GetPlugins::arguments const):
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments const):
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments const):
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments const):
(Messages::TestWithoutAttributes::TemplateTest::arguments const):
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments const):
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments const):
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments const):
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments const):
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments const):
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::stopMessageTesting):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPasteboardProxy.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::engineFailedToLoad):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioHardwareListener.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteImageDecoderAVFManager.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadRequestWaitingForProcessLaunch):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/257836@main
